src/Storages/StorageFuzzJSON.cpp

```diff
@@ -0,0 +1,563 @@
+#include <optional>
+#include <random>
+#include <string_view>
+#include <Columns/ColumnString.h>
+#include <Storages/StorageFactory.h>
+#include <Storages/StorageFuzzJSON.h>
+#include <Storages/checkAndGetLiteralArgument.h>
+#include <Common/JSONParsers/RapidJSONParser.h>
+#include <Common/JSONParsers/SimdJSONParser.h>
+#include <Common/checkStackSize.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+extern const int LOGICAL_ERROR;
+extern const int INCORRECT_DATA;
+}
+
+namespace
+{
+
+using uniform = std::uniform_int_distribution<size_t>;
+
+struct JSONNode;
+using JSONNodeList = std::list<std::shared_ptr<JSONNode>>;
+
+struct JSONValue
+{
+    enum class Type : size_t
+    {
+        Fixed = 0,
+        Array = 1,
+        Object = 2,
+    };
+
+    static Type getType(const JSONValue & v);
+
+    // The node value must be one of the following:
+    // Examples: 5, true, "abc"
+    std::optional<Field> fixed;
+    // Examples: [], ["a"], [1, true]
+    std::optional<JSONNodeList> array;
+    // Examples: {}, {"a": [1,2], "b": "c"}
+    std::optional<JSONNodeList> object;
+};
+
+JSONValue::Type JSONValue::getType(const JSONValue & v)
+{
+    if (v.fixed)
+    {
+        assert(!v.array);
```

Comment:
We also have `chassert` that works not only in debug, but in sanitizer builds. It's not that needs to be changed, but just to let you know
https://github.com/ClickHouse/ClickHouse/blob/f960036d76375eedee741f963be1dae313a65b7f/base/base/defines.h#L109-L116